src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -82,7 +82,9 @@ public function warmUp($cacheDir)
             return;
         }
 
-        foreach ($this->resourceLocales as $locale) {
+        $locales = array_merge(array($this->getLocale()), $this->getFallbackLocales(), $this->resourceLocales);
+        foreach (array_unique($locales) as $locale) {
+            $this->catalogues = array();
```

Comment:
but the loading of catalogues is loading their fallbacks to include them in the cache file. so this will load the translation of fallback locales many times (once per locale using them as fallback, plus for themselves).